critiquebrainz/frontend/external/spotify.py

```diff
@@ -6,12 +6,55 @@
 import requests
 import urllib.parse
 from brainzutils import cache
+from base64 import b64encode
+from flask import current_app as app
 
 DEFAULT_CACHE_EXPIRATION = 12 * 60 * 60  # seconds (12 hours)
+ACCESS_TOKEN_EXPIRATION = 60 * 60 # seconds (1 hour)
 
 BASE_URL = "https://api.spotify.com/v1"
 
 
+def fetch_access_token():
+    """Get an access token from the oauth credentials."""
+
+    key = cache.gen_key("spotify_access_token")
+    namespace = "spotify_access_token"
+    access_token = cache.get(key, namespace)
+
+    client_id = app.config.get("SPOTIFY_CLIENT_ID")
+    client_secret = app.config.get("SPOTIFY_CLIENT_SECRET")
+    auth_value = b64encode(bytes(f"{client_id}:{client_secret}", "utf-8")).decode("utf-8")
+
+    if not access_token:
+        access_token = requests.post(
+            "https://accounts.spotify.com/api/token",
+            data={"grant_type": "client_credentials"},
+            headers={"Authorization": f"Basic {auth_value}"},
+        ).json()
+        access_token = access_token.get('access_token')
+        if not access_token:
+            raise Exception("Could not fetch access token for Spotify")
+        cache.set(key=key, namespace=namespace, val=access_token, time=ACCESS_TOKEN_EXPIRATION)
+    return access_token
+
+
+def get_spotify(url):
+    """Make a GET request to Spotify Web API.
```

Comment:
This is not very accurate. It currently makes a GET request to whatever URL is provided, not just from spotify.com. One thing you can do here is instead accept a path with query string and append it to the base URL of the Spotify API.